torchaudio/csrc/CMakeLists.txt

```diff
@@ -155,20 +154,11 @@ if (BUILD_TORCHAUDIO_PYTHON_EXTENSION)
 
   target_link_libraries(
     _torchaudio
-    torch
```

Comment:
Is it okay to remove pytorch?

Reply:
`torch` is now linked to `libtorchaudio` from this [change](https://github.com/pytorch/audio/pull/1752/files#diff-12a57bc8e7ba6a5dd32aa566c805dd6c6514f46d26d5f53d0bf4400d557eaecbR67) so this should be okay

Reply:
Yup so the library dependency graph looks like the following

![Untitled](https://user-images.githubusercontent.com/855818/133638452-50f0f2a0-26d7-4d2c-b201-0451c5518b69.png)

Not sure if we need to explicitly add `libtorch` and `libsox` to `_torchaudio` in this case, because they will be handled either way.